runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaDoFnRunners.java

```diff
@@ -213,31 +218,65 @@ public void processElement(WindowedValue<InT> elem) {
               }
             };
 
-        try (RemoteBundle bundle =
+        remoteBundle =
             stageBundleFactory.getBundle(
                 receiverFactory,
                 StateRequestHandler.unsupported(),
-                BundleProgressHandler.ignored())) {
-          Iterables.getOnlyElement(bundle.getInputReceivers().values()).accept(elem);
-        }
+                BundleProgressHandler.ignored());
 
-        // RemoteBundle close blocks until all results are received
-        KV<String, FnOutT> result;
-        while ((result = outputQueue.poll()) != null) {
-          outputManager.output(
-              idToTupleTagMap.get(result.getKey()), (WindowedValue) result.getValue());
-        }
+        inputReceiver = Iterables.getOnlyElement(remoteBundle.getInputReceivers().values());
```

Comment:
I guess this is to say we only support main output right now? Maybe worth a comment for this.

Reply:
Igor already has a RB for side output. I think this is now addressed